src/tests/test_ticker_request.py

```diff
@@ -1,39 +1,147 @@
+import os
 import unittest
+from unittest.mock import patch, MagicMock
 
-from src.stac_utils.ticker_request import TickerRequest
+import requests
+
+from src.stac_utils.ticker_request import (
+    TickerRequest,
+    TickerException,
+    TickerAuthException,
+)
 
 
 class TestTickerRequest(unittest.TestCase):
+    def setUp(self) -> None:
+        self.test_auth = {
+            "TICKER_URL": "https://www.foo.bar",
+            "AUTH_USER": "spam",
+            "AUTH_PASS": "spam1234",
+        }
+
     def test_init(self):
         """Test init"""
+        with patch.dict(os.environ, values=self.test_auth):
+            TickerRequest()
 
     def test_init_no_auth(self):
         """Test init with missing auth"""
 
+        self.assertRaises(TickerAuthException, TickerRequest)
+
     def test_create_session(self):
         """Test create session"""
 
+        with patch.dict(os.environ, values=self.test_auth):
+            test_ticker = TickerRequest()
+            test_session = test_ticker.create_session()
+            self.assertIsInstance(test_session, requests.Session)
+            self.assertEqual(
+                test_session.auth,
+                (self.test_auth["AUTH_USER"], self.test_auth["AUTH_PASS"]),
+            )
+
     def test_transform_response(self):
         """Test transform response"""
 
+        test_response = requests.Response()
+        with patch.dict(os.environ, values=self.test_auth):
+            test_ticker = TickerRequest()
+            self.assertIs(test_response, test_ticker.transform_response(test_response))
+
     def test_check_for_error_when_exists(self):
         """Test check for error when one exists"""
 
+        test_response = requests.Response()
+        test_response.status_code = 500
+
+        with patch.dict(os.environ, values=self.test_auth):
+            test_ticker = TickerRequest()
+            self.assertRaises(
+                TickerException, lambda: test_ticker.check_for_error(test_response, {})
+            )
+
     def test_check_for_error_when_not_exists(self):
         """Test check for error when one does not exist"""
 
+        test_response = requests.Response()
+        test_response.status_code = 200
+
+        with patch.dict(os.environ, values=self.test_auth):
+            test_ticker = TickerRequest()
+            test_ticker.check_for_error(test_response, {})
+
     def test_add_data(self):
         """Test add data"""
 
+        with patch.dict(os.environ, values=self.test_auth):
+            test_ticker = TickerRequest()
+            test_data = {
+                "state": "UNK",
```

Comment:
for the sake of conformation with the actual BQ table which has a 2 character limit on this column, can you use "ZZ" or something